src/dsp_tools/commands/xmlupload/models/ontology_lookup_models.py

```diff
@@ -75,3 +77,78 @@ def _get_all_properties_from_json(onto_json: list[dict[str, Any]]) -> list[str]:
 
 def _remove_prefixes(ontology_elements: list[str]) -> list[str]:
     return [x.split(":")[1] for x in ontology_elements]
+
+
+AllowedEncodings = Literal["utf8", "xml"]
+
+
+@dataclass
+class TextValueData:
+    resource_id: str
+    property_name: str
+    encoding: AllowedEncodings
+
+
+@dataclass
+class PropertyTextValueEncodingTypes:
+    """
+    This class contains the information
+    which properties have which type of encoding for a TextValue property in the ontology.
+    """
+
+    formatted_text: set[str] = field(default_factory=set)
+    unformatted_text: set[str] = field(default_factory=set)
+
+
+def get_text_value_properties_and_formatting_from_json(
+    onto_json_dict: dict[str, list[dict[str, Any]]], default_onto: str
+) -> PropertyTextValueEncodingTypes:
+    """
+    This function takes a dict with the names and ontology json of the project.
+    It filters out the property with TextValues as object
+    and separates them into properties with formatted and unformatted text
+
+    Args:
+        onto_json_dict: dict with the project ontologies
+        default_onto: name of the default ontology
+
+    Returns:
+        Look-up containing the properties separated according to the formatting
+    """
+    all_props = []
+    for onto_json in onto_json_dict.values():
+        all_props.extend(_get_all_text_value_properties_and_types_from_json(onto_json))
+    return _make_text_value_property_type_lookup(all_props, default_onto)
+
+
+def _make_text_value_property_type_lookup(
+    prop_list: list[tuple[str, str]], default_onto: str
+) -> PropertyTextValueEncodingTypes:
+    re_pat = r"^" + default_onto + r":" + r".+$"
+
+    def remove_default_prefix(prop_str: str) -> str:
+        if regex.search(re_pat, prop_str) is not None:
```

Comment:
I don't think you need a variable for this, you can use the pattern directly.
Plus, `is not None` is more verbose than necessary:

```suggestion
        if not regex.search(f"^{default_onto}:.+$", prop_str):
```

Reply:
I will make it less complicated, but the `not` inverses the logic.